### DISCUSSION TITLE
JupyterLab support for NNI

### DISCUSSION THREAD

**Discussion by scarlett2018** [microsoft/nni#3668] [New Feature Design Discussion]
**Scenario** Aiming to facilitate Cloud (Azure, AliCloud, etc.) Linux VM as a data scientist's sandbox environment. 
**Env**: Linux VM with NNI + JupyterLab preinstalled docker
**Story**: As a data scientist using the above **Env**, I can embed NNI web UI with JupyterLab within the JupyterLab Tab so that I can have a unique debugging/programming experience inside JupyterLab with NNI.

@liuzhe-lz @weidankong @johnsonms

reference: https://github.com/chaoleili/jupyterlab_tensorboard

**Comment by JunweiSUN** [comment]
I think docker is hard to master for data scientists. Also, there may be some port issues if using docker. Can we embed JupyterLab in NNI's package and provide a command to start JupyerLab in NNI's codedir? (Maybe also a magic command in JupyterLab to start an experiment in a notebook.)

**Reply by mingaliu** [reply]
Yeah, instead of docker, we can have a Jupyter notebook server extension (and some UI nbextension).  NNI package can include those extension and we can have the NNI webUI directly render as a tab in notebook?
